nexus/lib/qmcpack_input.py

```diff
@@ -5254,6 +5255,57 @@ def generate_jastrow3(function='polynomial',esize=3,isize=3,rcut=4.,coeff=None,i
 #end def generate_jastrow3
 
 
+def generate_kspace_jastrow(kc1, kc2, nk1, nk2,
+  symm1='isotropic', symm2='isotropic', coeff1=None, coeff2=None):
+  """ generate <jastrow type="kSpace">
+
+  Args:
+    kc1 (float): kcut for one-body Jastrow
+    kc2 (float): kcut for two-body Jastrow
+    nk1 (int): number of coefficients for one-body Jastrow
+    nk2 (int): number of coefficients for two-body Jastrow
+    symm1 (str, optional): one of ['crystal', 'isotropic', 'none'],
+     default is 'isotropic'
+    symm2 (str, optional): one of ['crystal', 'isotropic', 'none'],
+     default is 'isotropic'
+    coeff1 (list, optional): one-body Jastrow coefficients, default None
+    coeff2 (list, optional): two-body Jastrow coefficients, default None
+  Returns:
+    QIxml: kspace_jastrow qmcpack_input element
+  """
+
+  if coeff1 is None: coeff1 = [0]*nk1
+  if coeff2 is None: coeff2 = [0]*nk2
+  if len(coeff1) != nk1: raise RuntimeError('coeff1 mismatch')
+  if len(coeff2) != nk2: raise RuntimeError('coeff2 mismatch')
```

Comment:
Please use the standard error handling mechanisms in Nexus.  In this case, `QmcpackInput.class_error('coeff2 mismatch','generate_kspace_jastrow')` would do fine.  These errors can all be converted to raise type errors under the hood (as nxs-test does) while defaulting to "abort with traceback" for typical use.